python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -0,0 +1,292 @@
+        """Create a chat message content object.
+
+        Args:
+            response: The response from the service.
+            choice: The choice from the response.
+            metadata: The metadata from the response.
+
+        Returns:
+            A chat message content object.
+        """
+        items = []
+        if choice.message.tool_calls:
+            for tool_call in choice.message.tool_calls:
+                items.append(
+                    FunctionCallContent(
+                        id=tool_call.id,
+                        name=tool_call.function.name,
+                        arguments=tool_call.function.arguments,
+                    )
+                )
+
+        return ChatMessageContent(
+            role=AuthorRole(choice.message.role),
+            items=items,
+            content=choice.message.content,
+            inner_content=response,
+            finish_reason=FinishReason(choice.finish_reason) if choice.finish_reason else None,
+            metadata=metadata,
+        )
+
+    def _create_streaming_chat_message_content(
+        self,
+        chunk: AsyncStreamingChatCompletions,
+        choice: StreamingChatChoiceUpdate,
+        metadata: dict[str, Any],
+    ) -> StreamingChatMessageContent:
+        """Create a streaming chat message content object.
+
+        Args:
+            chunk: The chunk from the response.
+            choice: The choice from the response.
+            metadata: The metadata from the response.
+
+        Returns:
+            A streaming chat message content object.
+        """
+        items = []
+        if choice.delta.tool_calls:
+            for tool_call in choice.delta.tool_calls:
+                items.append(
+                    FunctionCallContent(
+                        id=tool_call.id,
+                        index=choice.index,
+                        name=tool_call.function.name,
+                        arguments=tool_call.function.arguments,
+                    )
+                )
+
+        return StreamingChatMessageContent(
+            role=AuthorRole(choice.delta.role) if choice.delta.role else AuthorRole.ASSISTANT,
+            items=items,
+            content=choice.delta.content,
+            choice_index=choice.index,
+            inner_content=chunk,
+            finish_reason=FinishReason(choice.finish_reason) if choice.finish_reason else None,
+            metadata=metadata,
+        )
+
+    def _format_chat_history(self, chat_history: ChatHistory) -> list[ChatRequestMessage]:
+        """Format the chat history to the expected objects for the client.
+
+        Args:
+            chat_history: The chat history.
+
+        Returns:
+            A list of formatted chat history.
+        """
+        chat_request_messages: list[ChatRequestMessage] = []
+
+        for message in chat_history.messages:
+            if message.role == AuthorRole.USER and any(isinstance(item, ImageContent) for item in message.items):
```

Comment:
you could inverse this logic and then call append and continue, less indentation on the parsing of the content which makes it easier to read